android/src/toga_android/widgets/textinput.py

```diff
@@ -52,6 +64,10 @@ def set_alignment(self, value):
     def set_value(self, value):
         self.native.setText(value)
 
+    def set_on_return(self, handler):
+        self._ReturnListener = TogaReturnListener(self)
+        self.native.setOnKeyListener(self._ReturnListener)
```

Comment:
We've been trying to move away from "set_on_" handlers, instead preferring to install the native handler/listener as part of the widget setup.